doc/porting_guide.md

```diff
@@ -53,7 +53,7 @@ Porting the Xively C Client to new platforms is accelerated by the Board Support
 
 ## Board Support Package (BSP)
 
-The Board Support Package (BSP) is the well-defined set of functions that the Xively Client invokes to interact with a platform's specific networking, [TLS](https://en.wikipedia.org/wiki/Transport_Layer_Security), memory, random number generator, and time implementations.
+The Board Support Package (BSP) is the well-defined set of functions that the Xively Client invokes to interact with a platform's specific networking, [TLS](https://en.wikipedia.org/wiki/Transport_Layer_Security), memory, random number generator, file/firmware storage, and time implementations.
```

Comment:
I would separate these here too. Also I'd follow the following order: networking, file io, memory, random number generator, time implementation and firmware management.